packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -898,7 +914,26 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
         ignoring: !enabled,
         child: Container(
           decoration: effectiveDecoration,
-          child: _addTextDependentAttachments(paddedEditable, textStyle, placeholderStyle),
+          child: TextSelectionGestureDetector(
```

Comment:
The Align widget is what does the vertical alignment.  I had to move the TextSelectionGestureDetector here in order to get the entire input to be interactive in all cases like it should be.